README.md
# cncta-script-collection
7evens CNCTA Script Collection


CHANGELOG
2019-07-02 - Version: 1.2
-------------------------

- forked CnCTA SoO SCRIPT PACK CRX 1.3.2 from https://www.crx4chrome.com/crx/87686/
  and merged own changes


2019-05-19 - Version: 1.1.1
---------------------------

- updated script MaelstromTools Dev (Modv1.5 for MCV) to disable MCV popup by default
  (in favor of Next MCV Info HUD)
- updated script Next MCV Info HUD 1.0 to beta


2019-05-19 - Version: 1.1
-------------------------

- new script: Next MCV Info HUD 1.0 alpha


2019-05-11 - Version: 1.0
-------------------------

- Created fork from https://addons.opera.com/de/extensions/details/cncta-soo-script-pack/

- Updated scripts:
    - infernal wrapper 0.390737.7
    - Tiberium Alliances Battle Simulator 17.06.07
    - Tiberium Alliances Combat Simulator - TACS 3.53b
    - C&C TA View Player Base 1.1
    - Tiberium Alliances Tunnel Info 2.3
    - New Custom Flunik Tools (SaberTooths-Fix-for-Awesome) 1.4.3
    - Maelstrom ADDON Basescanner 1.9.0.B
    - Tiberium Alliances Formation Saver 2.3
    - Tiberium Alliances ReplayShare 0.4.3
    - WarChiefs - Tiberium Alliances Upgrade Base/Defense/Army 2017.06.06
    - PluginsLib - mhNavigator - Tiberium Alliances 1.36
    - Tiberium Alliances Report Summary 14.06.16
    - Tiberium Alliances - New Resource Trade Window 1.5.1
    - C&C: Tiberium Alliances Chat Helper Enhanced Yiannis Mod 3.2.1
    - OBSOLETE ~Bug Fixed > TAMarkerFix 0.11

- Added scripts:
    - C&C Tiberium Alliances Landingpage Autologin 1.0.1DL

- Fixed scripts:
    - C&C Tiberium Alliances POIs Analyser 2.0.2+ [fixed script error]
    - Tiberium Alliances Info Sticker 1.11.10+ [fixed script error]
    - C&C Tiberium Alliances Info Sticker (SUPERCompact) 1.11.10+ [fixed script error]
    - WarChiefs - Tiberium Alliances Sector HUD 13.12.18+ [fixed script error]
    - Shockr - Tiberium Alliances Tools - reMod 1.2 2.7.2.29+ [removed error, but there are more of them...]
    - PluginsLib - mhLoot - Tiberium Alliances 2.06+ [fixed broken images]
    - CnC: MHTools Tiberium Alliances Available Loot Summary + Info 1.8.3+ [fixed broken images]
    - MaelstromTools Dev (Modv1.5 for MCV) 0.1.4.6+ [fixed mcv popup jumping out of visibility]

- Modified scripts:
    - PluginsLib - mhNavigator - Tiberium Alliances 1.36+ [modified init to start minimized]


js/background.js
var DEFAULT_SCRIPTS = [ {
	id : 10019,
	greasyfork: 14532,
	name : "BaseInfo",
	cat : "infotool",
	hot: true,
	version : "  3.2.8",
	enabled : true
}, {
	id : 10029,
	greasyfork: 11112,
	name : "C&C: TA Hotkeys",
	cat : "tool",
	version : "  2.2.4",
	enabled : false
}, {
	id : 10030,
	greasyfork: 12791,
	name : "C&C TA Chat Colorize",
	cat : "gui",
	version : "  0.1.6",
	enabled : false
}, {
	id : 10049,
	greasyfork: 16009,
	name : "C&C TA Compass - ALT (points from the center of the screen)",
	cat : "gui",
	unique: 2,
	version : "  1.0.1",
	enabled : false
},{
	id : 10073,
	greasyfork: 30251,
	name : "CENTER DRIVEN Compass - Tiberium Alliances",
	cat : "gui",
	unique: 2,
	version : "  2.00",
	enabled : false
}, {
	id : 10048,
	greasyfork: 13925,
	name : "New Custom Flunik Tools (SaberTooths-Fix-for-Awesome)",
	cat : "tool",
	version : "  1.3.3.1",
	enabled : false
}, {
	id : 10015,
	greasyfork: 11647,
	name : "C&C TA View Player Base",
	cat : "infotool",
	version : "  1.1",
	enabled : false
}, {
	id : 10012,
	greasyfork: 670,
	name : "C&C TA CNCOpt Link Button",
	cat : "infotool",
	version : "  1.7.7",
	enabled : false
}, {
	id : 10045,
	greasyfork: 4192,
	name : "C&C TA CNCOpt Link Button for SC",
	cat : "infotool",
	hot: true,
	version : "  1.7.7b",
	enabled : false
}, {
	id : 10040,
	greasyfork: 1998,
	name : "C&C Tiberium Alliances Map",
	cat : "gui",
	version : "  15.11.28",
	enabled : false
}, {
	id : 10039,
	greasyfork: 716,
	name : "C&C Tiberium Coord Box Shortcut",
	version : "  1.2",
	cat : "gui",
	enabled : false
}, {
	id : 10014,
	greasyfork: 1934,
	name : "C&C Tiberium Alliances POIs Analyser",
	cat : "infotool",
	version : "  2.0.3",
	enabled : true
}, {
    id : 10018,
	greasyfork: 1856,
 	name : "C&C Tiberium Alliances MultiSession MOD by Ghostleader1402",
	cat : "login",
	version : "  0.6.1",
	enable : false
}, {
	id : 10016,
	greasyfork: 3343,
	name : "C&C TA Count Forgotten Bases in Range",
	cat : "infotool",
	version : "  1.1.0",
	enabled : false
}, {
	id : 10011,
	greasyfork: 1904,
	name : "C&C Tiberium Alliances Coords Button All",
	cat : "tool",
	version : "  2.0.2",
	enabled : false
}, {
	id : 10044,
	greasyfork: 1902,
	name : "C&C: Tiberium Alliances Chat Helper Enhanced Yiannis Mod",
	cat : "mail",
	hot: true,
	version : "  3.2.1",
	enabled : false
}, {
	id : 10047,
	greasyfork: 10382,
	name : "C&C: Tiberium Alliances - xTr1m's Base Overlay- Display Ratio 4:3",
	cat : "gui",
	version : "  1.13",
	enabled : false
}, {
	id : 10034,
	greasyfork: 715,
	name : "C&C MHTools Tiberium Alliances Available Loot Summary + Info",
	cat : "infotool",
	hot: true,
	version : "  1.8.3+",
	enabled : false
}, {
	id : 10041,
	greasyfork: 13916,
	name : "CityMoveInfoExtend",
	cat : "infotool",
	version : "  14.06.17",
	enabled : true
}, {
	id : 10042,
	greasyfork: 1901,
	name : "Command & Conquer TA Zmail",
	cat : "mail",
	version : "  2015.11.08",
	enabled : false
}, {
	id : 10001,
	greasyfork: 13898,
	name : "infernal wrapper",
	cat : "wrapper",
	hot: true,
	version : "  0.390737.7",
	enabled : true
}, {
	id : 10020,
	userscript: 162680,
	name : "KlickitlikeShing0",
	cat : "tool",
	version : "  0.993",
	enabled : false
}, {
	id : 10006,
	greasyfork: 13713,
	name : "MaelstromTools Dev (Modv1.6 for MCV)",
	cat : "tool",
	hot: true,
	version : "  0.1.4.7+",
	enabled : true
}, {
	id : 10008,
	greasyfork: 13714,
	name : "Maelstrom ADDON Basescanner Basic",
	cat : "tool",
	unique: 1,
	hot: true,
	version : "  1.9.0.B",
	enabled : false
},{
	id : 10069,
	name : "Maelstrom ADDON Basescanner + cncoptplus growth rate",
	info: "http://chertosha.com/cncoptplus.html",
	cat : "tool",
	unique: 1,
	version : "  1.9.0.B",
	enabled : false
},{
	id : 10071,
	greasyfork: 19672,
	name : "Maelstrom ADDON Basescanner Plus Infected Camps",
	cat : "tool",
	unique: 1,
	version : "  1.9.0.I",
	enabled : true
},{
	id : 10056,
	name : "Next MCV Info HUD",
	cat : "infotool",
	version : "  1.0 beta",
	enabled : true,
	hot : true,
}, {
	id : 10021,
	greasyfork: 10436,
	name : "Maelstrom ADDON City Online Status Colorer for SC",
	cat : "gui",
	hot: true,
	version : "  0.6.5",
	enabled : false
}, {
	id : 10052,
	userscript: 482069,
	name : "Massive DynamiX Attack Range",
	version : "  2.0",
	enabled : true
}, {
	id : 10050,
	userscript: 160800,
	name : "PluginsLib - mhLoot - Tiberium Alliances",
	cat : "infotool",
	version : "  2.06",
	enabled : false
}, {
	id : 10010,
	greasyfork: 1860,
	name : "PluginsLib - mhNavigator - Tiberium Alliances",
	cat : "gui",
	hot: true,
	version : "  1.35",
	enabled : true
}, {
	id : 10051,
	greasyfork: 24754,
	name : "Tiberium Alliances Report Summary",
	cat: "gui",
	version : "  17.06.06",
	enabled : true
}, {
	id : 10043,
	greasyfork: 17032,
	name : "Shockr - Tiberium Alliances Tools - BaseScanner MailVersion- reMod 1.3",
	cat : "tool",
	hot: true,
	version : "  2.7.2.29+",
	enabled : true
}, {
	id : 10038,
	greasyfork: 13802,
	name : "The Green Cross - Tiberium Alliances Tools",
	cat : "tool",
	hot: true,
	version : "  0.61",
	enabled : true
}, {
	id : 10046,
	userscript: 154546,
	name : "The Green Cross - Tiberium Alliances Combat Simulator",
	cat : "simulator",
	version : "  3.2.2",
	enabled : false
}, {
	id : 10022,
	greasyfork: 11916,
	name : "Tiberium Alliances Attack Range",
	version : "  2.2",
	cat : "tool",
	enabled : true
}, {
	id : 10028,
	greasyfork: 13711,
	name : "Tiberium Alliances Battle Simulator - V2",
	cat : "simulator",
	hot: true,
	version : "  18.07.12+",
	enabled : true
}, {
	id : 10000,
	name : "Tiberium Alliances Combat Simulator - TACS",
	cat : "simulator",
	version : "  3.53b",
	greasyfork: 14830,
	enabled : false
}, {
	id : 10009,
	greasyfork: 1905,
	name : "Tiberium Alliances Formation Saver",
	cat : "tool",
	hot: true,
	version : "  2.3",
	enabled : true
}, {
	id : 10033,
	greasyfork: 30450,
	name : "Tiberium Alliances Info Sticker",
	version : "  1.11.10.2++",
	hot: true,
	unique: 3,
	cat : "infotool",
	enabled : false
}, {
	id : 10063,
	name : "Tiberium Alliances Info Sticker (SUPERCompact)",
	cat : "infotool",
	version : "  1.11.10.1+",
	unique: 3,
	enabled : false
}, {
	id : 10002,
	greasyfork: 1936,
	name : "C&C Tiberium Alliances PvP/PvE Ranking, POI Holding and split base kill score",
	cat : "infotool",
	hot: true,
	version : "  1.7.3",
	enabled : true
}, {
	id : 10007,
	greasyfork: 1933,
	name : "Tiberium Alliances - New Resource Trade Window",
	cat : "gui",
	hot: true,
	version : "  1.5.1",
	enabled : false
}, {
  id: 10079,
  name: "Tiberium Alliances Transfer All Resources -KRS_L-",
	cat : "tool",
  version: "  1.6.1",
  enabled: false,
	hot: true
}, {
	id : 10054,
	greasyfork: 18248,
	name : "Tiberium Alliances Repair Time Of Death",
	cat : "gui",
	version : "  1.0.1",
	enabled : false
}, {
	id : 10025,
	greasyfork: 18246,
	name : "Tiberium Alliances ReplayShare",
	cat : "tool",
	hot: true,
	version : "  0.4.1",
	enabled : true
}, {
	id : 10026,
	greasyfork: 18245,
	name : "Tiberium Alliances Report Stats",
	cat : "infotool",
	version : "  0.5.3",
	enabled : false
}, {
	id : 10023,
	greasyfork: 10438,
	name : "Tiberium Alliances Real POI Bonus",
	cat : "infotool",
	hot: true,
	version : "  1.0.2",
	enabled : true
}, {
	id : 10004,
	greasyfork: 30401,
	name : "Tiberium Alliances Supplies Mod",
	cat : "gui",
	version : "  1.41",
	enabled : false
}, {
	id : 10053,
	greasyfork: 18079,
	name : "Tiberium Alliances The Movement",
	cat : "infotool",
	version : "  1.0.3.2",
	enabled : true,
	hot: true
}, {
	id : 10003,
	greasyfork: 29201,
	name : "Tiberium Alliances Tunnel Info",
	version : "  2.2.1.2",
	hot: true,
	cat : "infotool",
	enabled : true
}, {
	id : 10037,
	greasyfork: 10440,
	name : "Tiberium Alliances Tweaks",
	version : "  1.2.0",
	hot: true,
	cat : "tweak",
	enabled : true
}, {
	id : 10005,
	userscript: 204128,
	name : "Tiberium Alliances URL Bug Fix",
	version : "  1.1",
	cat : "tweak",
	enabled : false
}, {
	id : 10027,
	greasyfork: 18247,
	name : "Tiberium Alliances Wavy",
	cat : "infotool",
	hot: true,
	version : "  0.5.5",
	enabled : true
}, {
	id : 10032,
	greasyfork: 1994,
	name : "WarChiefs - Tiberium Alliances Combat Simulator",
	cat : "simulator",
	version : "  13.09.26",
	enabled : false
}, {
	id : 10036,
	greasyfork: 30634,
	name : "WarChiefs - Tiberium Alliances Sector HUD",
	cat : "gui",
	hot: true,
	version : "  13.12.18.1",
	enabled : true
}, {
	id : 10017,
	greasyfork: 1986,
	name : "WarChiefs - Tiberium Alliances Upgrade Base/Defense/Army",
	cat : "tool",
	hot: true,
	version : "  2017.06.06",
	enabled : true
}, {
	id : 10060,
	greasyfork: 17351,
	name : "TAMarkerFix",
	cat : "tweak",
	hot: false,
	version : "  0.11",
	enabled : false
}, {
	id : 10061,
	greasyfork: 20148,
	name : "TAMailMessageErrorFix",
	cat : "tweak",
	hot: true,
	version : "  0.2",
	enabled : false
}, {
	id : 10062,
	name : "MENU FIX Chrome 55",
	cat : "tweak",
	info: "http://forum.alliances.commandandconquer.com/showthread.php?tid=49151&pid=234961#pid234961",
	hot: true,
	version : "  1.1",
	enabled : true
},{
	id : 10064,
	name : "CnC TA: Crucial Script CNC-Map.com link",
	cat : "gui",
	lib : "parse",
	version : "  1.0.10",
	enabled : false
},{
	id : 10065,
	name : "AllianceMemberOnline",
	cat : "infotool",
	version : "  0.1.3.0",
	enabled : false
},{
	id : 10066,
	name : "Tiberium Alliances Auto Repair",
	cat : "tool",
	version : "  0.2.3",
	enabled : false,
	hot: true
},{
	id : 10067,
	name : "Tiberium Alliances Upgrade Top (ModButtonPos)",
	info: "https://openuserjs.org/scripts/tb23911/Tiberium_Alliances_Upgrade_Top",
	cat : "tool",
	version : "  0.9.0",
	enabled : false
},{
	id : 10068,
	name : "Tiberium Alliances Zoom",
	greasyfork: 11149,
	hot: true,
	cat : "gui",
	version : "  1.0.1",
	enabled : false
},
{
	id : 10072,
	greasyfork: 4089,
	name : "CENTER DRIVEN PvP Quick Map -XDaast-",
	version : "  1.5.2b",
	cat : "infotool",
	enabled : true,
	hot: true
},
{
  id: 10076,
	greasyfork: 30611,
  name: "CENTER DRIVEN Player Base Info (Basic)",
	cat : "infotool",
  version: "  3.00",
  enabled: true,
	hot: true
},
{
  id: 10080,
	greasyfork: 10434,
  name: "Tiberium Alliances BaseNavBar Reorderer",
	cat : "gui",
  version: "  1.0",
  enabled: false
},
{
  id: 10074,
  name: "CENTER DRIVEN PvP Alert Status -XDaast-",
  version: "  1.00",
  enabled : false,
	cat : "infotool",
	greasyfork: 30224,
	hot: true
},
{
  id: 10075,
  name: "CnC TA 'Auto-Login' (+ Auto repair + Auto packet collect + Auto reload)",
  version: "  0.27",
  enabled : false,
	cat : "tool",
	greasyfork: 36231,
	hot: false
},
{
  id: 10077,
  name: "leoStats",
  version: "  2018.06.05",
  enabled : false,
	cat : "infotool",
	info: "https://leostats.000webhostapp.com/",
	hot: false
},
{
  id: 10078,
  name: "MHTools Tiberium Alliances Available Loot Summary + Info",
  version: "  1.8.3.2",
  enabled : true,
	cat : "infotool",
	greasyfork: 715,
	hot: false
}
];

var updateopen = false;


	function openupdate()  {
		if (updateopen === true){
       chrome.tabs.create({
       url: chrome.runtime.getURL('updated.html')
    });
       updateopen = false;
	   }
       }




if (chrome && chrome.runtime && chrome.runtime.onInstalled) {
	chrome.runtime.onInstalled.addListener(function(){
	updateopen = true;
	openupdate();

});

} else {

	updateopen = true;
}

var CURRENT_VERSION = chrome.runtime.getManifest().version;
var PREVIOUS_VERSION = localStorage.getItem('CNCTA_VERSION');
if (CURRENT_VERSION !== PREVIOUS_VERSION) {
	if (updateopen){
	openupdate();
	}
	localStorage.setItem('CNCTA_VERSION', CURRENT_VERSION);
	localStorage.setItem('CNCTA_SCRIPTS', JSON.stringify(DEFAULT_SCRIPTS));


}
	var enabled = JSON.parse(localStorage.getItem('CNCTA_ENABLED'));
	if (enabled === null || typeof enabled !== 'object') {
		enabled = {};
		var tmp = {};

	for (var i in DEFAULT_SCRIPTS) {
		var script = DEFAULT_SCRIPTS[i];
		tmp[script.id] = enabled[script.id] || script.enabled;
	}
	localStorage.setItem('CNCTA_ENABLED', JSON.stringify(tmp));
	}


function processRequest(request, sender, sendResponse) {
	"use strict";
	switch (request.type) {
	case "get":
		var data = {};
		if (request.name.constructor == Array) {
			for ( var i = 0; i < request.name.length; i++) {
				data[request.name[i]] = localStorage[request.name[i]];
			}
			sendResponse(data);
		} else {
			sendResponse(localStorage[request.name]);
		}
		break;
	case "set":
		localStorage[request.name] = request.value;
		break;
	case "pageAction":
		chrome.pageAction.show(sender.tab.id);
		sendResponse({});
		break;
	default:
		console.log("invalid request " + request);
		break;
	}
}


chrome.runtime.onMessage.addListener(processRequest);


/*------------------------------------------------*/
chrome.pageAction.onClicked.addListener(function() {
	if (chrome.runtime.openOptionsPage) {
    // New way to open options pages, if supported (Chrome 42+).
    chrome.runtime.openOptionsPage();
  } else {
    // Reasonable fallback.
    window.open(chrome.runtime.getURL('options.html'));
  }
});
/*------------------------------------------------*/
